src/Avalonia.Input/KeyboardDevice.cs

```diff
@@ -70,66 +70,60 @@ public void ProcessRawEvent(RawInputEventArgs e)
+            {
+                switch (keyInput.Type)
                 {
-                    switch (keyInput.Type)
-                    {
-                        case RawKeyEventType.KeyDown:
-                        case RawKeyEventType.KeyUp:
-                            var routedEvent = keyInput.Type == RawKeyEventType.KeyDown
-                                ? InputElement.KeyDownEvent
-                                : InputElement.KeyUpEvent;
-
-                            KeyEventArgs ev = new KeyEventArgs
-                            {
-                                RoutedEvent = routedEvent,
-                                Device = this,
-                                Key = keyInput.Key,
-                                KeyModifiers = KeyModifiersUtils.ConvertToKey(keyInput.Modifiers),
-                                Source = element,
-                            };
-
-                            IVisual currentHandler = element;
-                            while (currentHandler != null && !ev.Handled && keyInput.Type == RawKeyEventType.KeyDown)
-                            {
-                                var bindings = (currentHandler as IInputElement)?.KeyBindings;
-                                if(bindings!=null)
-                                    foreach (var binding in bindings)
-                                    {
-                                        if(ev.Handled)
-                                            break;
-                                        binding.TryHandle(ev);
-                                    }
-                                currentHandler = currentHandler.VisualParent;
-                            }
-
-                            element.RaiseEvent(ev);
-                            e.Handled = ev.Handled;
-                            break;
-                    }
+                    case RawKeyEventType.KeyDown:
+                    case RawKeyEventType.KeyUp:
+                        var routedEvent = keyInput.Type == RawKeyEventType.KeyDown
+                            ? InputElement.KeyDownEvent
+                            : InputElement.KeyUpEvent;
+
+                        KeyEventArgs ev = new KeyEventArgs
+                        {
+                            RoutedEvent = routedEvent,
+                            Device = this,
+                            Key = keyInput.Key,
+                            KeyModifiers = KeyModifiersUtils.ConvertToKey(keyInput.Modifiers),
+                            Source = element,
+                        };
+
+                        IVisual currentHandler = element;
+                        while (currentHandler != null && !ev.Handled && keyInput.Type == RawKeyEventType.KeyDown)
+                        {
+                            var bindings = (currentHandler as IInputElement)?.KeyBindings;
+                            if (bindings != null)
+                                foreach (var binding in bindings)
```

Comment:
I know that you just moved this code around but - oh god why do we have such `if`  and `foreach` setup without brackets :)

Reply:
I think this is only fine when a single line follows but I prefer using brackets all the time.

Reply:
Ha, I know :) But yeah I didn't want to do any refactoring to keep the PR clean.